packages/storybook/src/generators/configuration/util-functions.ts

```diff
@@ -356,3 +356,16 @@ export function getTsConfigPath(
       : 'tsconfig.lib.json'
   );
 }
+
+export function addBuildStorybookToCacheableOperations(tree: Tree) {
+  updateJson(tree, 'nx.json', (json) => {
+    json.tasksRunnerOptions.default.options.cacheableOperations = Array.from(
+      new Set([
+        ...json.tasksRunnerOptions.default.options.cacheableOperations,
+        'build-storybook',
+      ])
+    );
```

Comment:
We need to add some null checks here, ideally this path would be available but tasksRunnerOptions has bitten us several times in the past. If nothing else, just something like:
```suggestion
    if ( json.tasksRunnerOptions?.default?.options?.cacheableOperations )
    json.tasksRunnerOptions.default.options.cacheableOperations = Array.from(
      new Set([
        ...json.tasksRunnerOptions.default.options.cacheableOperations,
        'build-storybook',
      ])
    );
```